javascript/modules/client/src/internal/client-core.ts

```diff
@@ -55,11 +62,18 @@ export abstract class ClientCore implements IClientCore {
     if (!signer) {
       throw new Error("Empty wallet or signer");
     }
-
     this._signer = signer;
     return this;
   }
 
+  public shiftIpfsNode() {
+    if (!this._ipfs.length) throw new Error("No endpoints");
```

Comment:
Might have not been initialized yet

```suggestion
    if (!this._ipfs?.length) throw new Error("No IPFS endpoints are available");
```